File: app/src/main/java/com/zhang/readme/view/base/BaseActivity.java
package com.zhang.readme.view.base;

import android.os.Bundle;
import android.support.annotation.LayoutRes;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;

import butterknife.ButterKnife;

/**
 * Created by zhang on 2017/3/30.
 *
 * Activity基类
 */

public abstract class BaseActivity extends AppCompatActivity {

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(bindLayout());
        ButterKnife.bind(this);

        initVar();
        initView();
    }

    @LayoutRes
    protected abstract int bindLayout();
    protected abstract void initVar();
    protected abstract void initView();

}

File: app/src/main/java/cn/devifish/readme/view/base/BaseActivity.kt
package cn.devifish.readme.view.base

import android.os.Bundle
import android.support.annotation.LayoutRes
import android.support.v7.app.AppCompatActivity
import android.view.WindowManager
import io.reactivex.disposables.CompositeDisposable

/**
 * Created by zhang on 2017/6/3.
 *
 */
abstract class BaseActivity: AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        if (bindLayout() != 0) setContentView(bindLayout())

        initVar()
        initView()
    }

    override fun onDestroy() {
        super.onDestroy()
    }

    @LayoutRes
    protected abstract fun bindLayout(): Int

    protected abstract fun initVar()
    protected abstract fun initView()

}